src/components/image/Image.base.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { useCallback, useEffect, useState } from 'react';
+
+import { PImage } from './Image.props';
+
+const ImageBase: React.FC<PImage> = ({ source, width, height, alt, className }: PImage) => {
+    if (!source) {
+        throw new Error('Compass Components: You need to provide image source');
+    }
+
+    const [loading, setLoading] = useState(true);
+    const [image, setImage] = useState('');
+
+    const fetchUrl = useCallback(async () => {
+        const response = await fetch(source);
```

Comment:
I prefer using `axios` for fetching. I don't exactly know if there is a difference, but a few years back `axios` was more performant than the built-in methods. Something to investigate!

Reply:
I can replace this with your util function